client/components/signup-form/index.jsx

```diff
@@ -311,13 +311,13 @@ export default React.createClass( {
 						autoCapitalize="off"
 						autoCorrect="off"
 						className="signup-form__input"
-						disabled={ this.state.submitting || this.props.disabled }
+						disabled={ this.state.submitting || this.props.disabled || ! this.props.allowEmailChange }
```

Comment:
Adding `! this.props.allowEmailChange` here assumes forces parent components to pass a truthy value. Also, this is breaking the signup form on the `/start/account/user` route for logged out users.